src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java

```diff
@@ -39,26 +38,39 @@ public static int getHearingDuration(SscsCaseData caseData, ReferenceDataService
             return overrideFields.getDuration().intValue();
         }
 
-        Integer duration = getHearingDurationAdjournment(caseData);
-        if (isNull(duration)) {
-            duration = getHearingDurationBenefitIssueCodes(caseData, referenceDataServiceHolder);
+        Integer duration = getHearingDurationAdjournment(caseData, referenceDataServiceHolder);
+        if (nonNull(duration)) {
+            return duration;
         }
 
-        return nonNull(duration) ? duration : DURATION_DEFAULT;
+        duration = getHearingDurationBenefitIssueCodes(caseData, referenceDataServiceHolder);
+        if (nonNull(duration)) {
+            return duration;
+        }
+
+        return DURATION_DEFAULT;
     }
 
-    public static Integer getHearingDurationAdjournment(SscsCaseData caseData) {
+    public static Integer getHearingDurationAdjournment(
+        SscsCaseData caseData,
+        ReferenceDataServiceHolder referenceDataServiceHolder
+    ) {
+        if (!referenceDataServiceHolder.isAdjournmentFlagEnabled()) {
+            return null;
```

Comment:
Integer can be null - int cannot as it is a primitive type. Optional would work here, but I think the current implementation is fine because immediately after the getHearingDurationAdjournment call there is a null check. So the method returning null is expected and desirable.